src/components/LoginPopup/LoginPopup.js

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import PopupWithForm from "../PopupWithForm/PopupWithForm";
+import './loginpopup.css';
+//import { Link } from "react-router-dom";
+
+function LoginPopup({isOpen, onClose, onRegisterClick, onLogin}) {
+
+  const [email, setEmail] = React.useState('');
+  const [password, setPassword] = React.useState('');
+  // const [userData, setUserData] = React.useState({});
+  // const [errors, setErrors] = React.useState({email: false, password: false});
```

Comment:
**Might be better**


If it’s interesting for you here is a custom `validation hook`.

You should put it in a separate file  `useFormAndValidation.js` in folder `hooks` and then import it to the component that needs validation:

```js

import {useState, useCallback} from 'react';

export function useFormAndValidation() {
  const [ values, setValues ] = useState({});
  const [ errors, setErrors ] = useState({});
  const [ isValid, setIsValid ] = useState(true);

  const handleChange = (e) => {
    const {name, value} = e.target
    setValues({...values, [name]: value });
    setErrors({...errors, [name]: e.target.validationMessage});
    setIsValid(e.target.closest('form').checkValidity());
  };



  const resetForm = useCallback((newValues = {}, newErrors = {}, newIsValid = false) => {
    setValues(newValues);
    setErrors(newErrors);
    setIsValid(newIsValid);
  }, [setValues, setErrors, setIsValid]);

  return { values, handleChange, errors, isValid, resetForm, setValues, setIsValid };
}


```

And only one line of code enables validation:

```js

  const {values, handleChange, errors, isValid, setValues, resetForm} = useFormAndValidation()

```

So, you **don’t need** to make functions for handling inputs, storing values, resetting the form. Everything is already done.